src/server/queue_recov_db.c

```diff
@@ -221,7 +231,42 @@ que_recov_db(char *qname, pbs_queue	*pq)
 
 		if (pque)
 			que_free(pque); /* free if we allocated here */
-		
 	}
 	return pq;
 }
+
+/**
+ * @brief
+ *	Refresh/retrieve queue from database and add it into AVL tree if not present
+ *
+ *	@param[in]	dbobj - The pointer to the wrapper queue object of type pbs_db_que_info_t
+ *	@param[out]	refreshed - To check if queues refreshed
+ *
+ * @return	The recovered queue
+ * @retval	NULL - Failure
+ * @retval	!NULL - Success, pointer to queue structure recovered
+ *
+ */
+pbs_queue *
+recov_queue_cb(pbs_db_obj_info_t *dbobj, int *refreshed) 
+{
+	pbs_queue *pque = NULL;
+	pbs_db_que_info_t *dbque = dbobj->pbs_db_un.pbs_db_que;
+
+	*refreshed = 0;
+	if ((pque = que_recov_db(dbque->qu_name, NULL)) != NULL) {
+		/* que_recov increments sv_numque */
+		sprintf(log_buffer, msg_init_recovque, pque->qu_qs.qu_name);
+		log_event(PBSEVENT_SYSTEM | PBSEVENT_ADMIN | PBSEVENT_DEBUG, PBS_EVENTCLASS_SERVER, LOG_INFO, msg_daemonname, log_buffer);
+		if (pque->qu_attr[(int) QE_ATR_ResourceAssn].at_flags & ATR_VFLAG_SET)
+			que_attr_def[(int) QE_ATR_ResourceAssn].at_free(&pque->qu_attr[(int) QE_ATR_ResourceAssn]);
```

Comment:
Why resource assigned is freed here? It is not getting decoded in decode_attr_db. Are we freeing it for server and node as well?

Reply:
If so, we should be seeing double counts with server and node resc assigned as well. Please confirm.